src/plugins/vis_type_xy/public/components/xy_threshold_line.tsx

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { FC } from 'react';
+
+import { AnnotationDomainTypes, LineAnnotation } from '@elastic/charts';
+
+import { ThresholdLineConfig } from '../types';
+
+type XYThresholdLineProps = ThresholdLineConfig & {
+  groupId?: string;
+};
+
+export const XYThresholdLine: FC<XYThresholdLineProps> = ({
+  show,
+  value: dataValue,
+  color,
+  width,
+  groupId,
+  dash,
+}) => {
+  if (!show) {
```

Comment:
Because of how we handle upserting specs within `@elastic/charts` any change in this component would trigger a full re-render of the chart regardless.

@markov00 correct me if I'm wrong here.

Reply:
oh good to know!